planning/behavior_path_planner/src/scene_module/pull_out/pull_out_module.cpp

```diff
@@ -617,150 +421,127 @@ bool PullOutModule::isInLane(
-  const double distance_to_goal_on_road_lane =
-    util::getSignedDistance(current_pose, goal_pose, lanelets);
-
-  return distance_to_goal_on_road_lane > pull_out_total_distance_min;
-}
+  const auto arclength_current =
+    lanelet::utils::getArcCoordinates(status_.current_lanes, current_pose);
+  const auto arclength_pull_out_end =
+    lanelet::utils::getArcCoordinates(status_.current_lanes, status_.pull_out_path.end_pose);
 
-bool PullOutModule::isSafe() const { return status_.is_safe; }
+  // has passed pull out end point
+  return arclength_current.length - arclength_pull_out_end.length >
+         parameters_.pull_out_finish_judge_buffer;
+}
 
-bool PullOutModule::isNearEndOfLane() const
+void PullOutModule::checkBackFinished()
 {
+  // check ego car is close enough to goal pose
   const auto current_pose = planner_data_->self_pose->pose;
-  const auto common_parameters = planner_data_->parameters;
-  const double threshold = 5 + common_parameters.minimum_pull_over_length;
+  const auto distance =
+    tier4_autoware_utils::calcDistance2d(current_pose, status_.pull_out_start_pose);
 
-  return std::max(0.0, util::getDistanceToEndOfLane(current_pose, status_.current_lanes)) <
-         threshold;
-}
+  const bool is_near = distance < parameters_.th_arrived_distance_m;
+  const double ego_vel = util::l2Norm(planner_data_->self_odometry->twist.twist.linear);
+  const bool is_stopped = ego_vel < parameters_.th_stopped_velocity_mps;
 
-bool PullOutModule::isCurrentSpeedLow() const
-{
-  const auto current_twist = planner_data_->self_odometry->twist.twist;
-  const double threshold_kmph = 10;
-  return util::l2Norm(current_twist.linear) < threshold_kmph * 1000 / 3600;
+  if (!status_.back_finished && is_near && is_stopped) {
+    RCLCPP_INFO(getLogger(), "back finished");
+    status_.back_finished = true;
+
+    // requst pull_out approval
+    waitApproval();
+    removeRTCStatus();
+    uuid_ = generateUUID();
+    // requset approval when stopped at the corresponding point, so distance is 0
+    updateRTCStatus(0.0);
+    current_state_ = BT::NodeStatus::SUCCESS;  // for breaking loop
+  }
 }
 
-bool PullOutModule::hasFinishedPullOut() const
+bool PullOutModule::isStopped()
 {
-  // check ego car is close enough to goal pose
-  const auto current_pose = planner_data_->self_pose->pose;
-  const auto arclength_current =
-    lanelet::utils::getArcCoordinates(status_.current_lanes, current_pose);
-  const auto arclength_shift_end =
-    lanelet::utils::getArcCoordinates(status_.current_lanes, status_.pull_out_path.shift_point.end);
-  const bool car_is_on_goal = (arclength_shift_end.length - arclength_current.length <
-                               parameters_.pull_out_finish_judge_buffer)
-                                ? true
-                                : false;
-
-  return car_is_on_goal;
+  odometry_buffer_.push_back(planner_data_->self_odometry);
+  // Delete old data in buffer
+  while (true) {
```

Comment:
rclcpp::ok()

Reply:
fixed in https://github.com/autowarefoundation/autoware.universe/pull/1438/commits/74f1feb7410fe50e81887d9563cbb9c24fe95751